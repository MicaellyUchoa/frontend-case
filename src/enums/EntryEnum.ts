export enum EntryEnum {
    CREDIT = 'CREDIT',
    DEBIT = 'DEBIT',
}
