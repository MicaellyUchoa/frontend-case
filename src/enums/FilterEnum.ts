export enum FilterEnum {
    ALL = 'Tudo',
    FUTURE = 'Futuro',
    EXIT = 'Sa√≠da',
    ENTRY = 'Entrada',
}
