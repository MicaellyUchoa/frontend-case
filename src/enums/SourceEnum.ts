export enum SourceEnum {
    PAYMENT = 'PAYMENT',
    TRANSFER = 'TRANSFER',
}
