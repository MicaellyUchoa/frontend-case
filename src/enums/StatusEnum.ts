export enum StatusEnum {
    COMPLETED = 'COMPLETED',
    REFUNDED = 'REFUNDED',
    PENDING = 'PENDING',
}
