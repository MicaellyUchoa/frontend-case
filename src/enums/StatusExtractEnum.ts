export enum StatusExtractEnum {
    ALL = 'Tudo',
    CREDIT = 'Entrada',
    DEBIT = 'Sa√≠da',
}
