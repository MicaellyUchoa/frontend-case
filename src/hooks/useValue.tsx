import { EntryEnum } from '../enums/EntryEnum';
import { StatusEnum } from '../enums/StatusEnum';
import formatMoney from '../utils/formatMoney';
import { useState, useEffect } from 'react';

interface useStatusIconProps {
    status: StatusEnum;
    entry: EntryEnum;
    value: number;
}
function useValue({ status, entry, value }: useStatusIconProps): JSX.Element {
    const handleStyleValue = (): string => {
        if (status === StatusEnum.REFUNDED) return 'text-c_grayscale line-through';
        switch (entry) {
            case EntryEnum.DEBIT:
                return 'text-c_primary';
            case EntryEnum.CREDIT:
                return 'text-c_secondary';
        }
    };

    const renderSignal = (): string | undefined => {
        if (status === StatusEnum.REFUNDED) return;
        switch (entry) {
            case EntryEnum.DEBIT:
                return ' - ';
            case EntryEnum.CREDIT:
                return ' + ';
        }
    };

    return (
        <p className={`${handleStyleValue()} sm:text-end`}>
            {renderSignal()}
            <span className={`${status !== StatusEnum.REFUNDED && 'font-black'}`}>{formatMoney(value)}</span>
        </p>
    );
}

export default useValue;
