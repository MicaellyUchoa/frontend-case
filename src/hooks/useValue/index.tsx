import { EntryEnum } from '../../enums/EntryEnum';
import { StatusEnum } from '../../enums/StatusEnum';
import formatMoney from '../../utils/formatMoney';
import { IUseValueProps } from '../../interfaces/IUseValueProps';

function useValue({ status, entry, value }: IUseValueProps): JSX.Element {
    //TODO test here
    const handleStyleValue = (): string => {
        if (status === StatusEnum.REFUNDED) return 'text-c_grayscale line-through';
        switch (entry) {
            case EntryEnum.DEBIT:
                return 'text-c_primary';
            case EntryEnum.CREDIT:
                return 'text-c_secondary';
        }
    };

    //TODO test here
    const renderSignal = (): string | undefined => {
        if (status === StatusEnum.REFUNDED) return;
        switch (entry) {
            case EntryEnum.DEBIT:
                return ' - ';
            case EntryEnum.CREDIT:
                return ' + ';
        }
    };

    return (
        <p className={`${handleStyleValue()} sm:text-end`}>
            {renderSignal()}
            <span className={`${status !== StatusEnum.REFUNDED && 'font-black'}`}>{formatMoney(value)}</span>
        </p>
    );
}

export default useValue;
