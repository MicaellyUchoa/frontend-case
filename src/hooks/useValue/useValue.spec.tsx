import { renderHook } from '@testing-library/react';
import { StatusEnum } from '../../enums/StatusEnum';
import useValue from './index';
import { EntryEnum } from '../../enums/EntryEnum';
describe('useValue Hook', () => {
    test('should call with status (refunded)', async () => {
        const { result } = renderHook(() =>
            useValue({
                status: StatusEnum.REFUNDED,
                entry: EntryEnum.CREDIT,
                value: 9000,
            }),
        );

        expect(result.current).toMatchSnapshot();
    });

    describe('useValue Hook (entry)', () => {
        test('should call with entry (debit)', async () => {
            const { result } = renderHook(() =>
                useValue({
                    status: StatusEnum.COMPLETED,
                    entry: EntryEnum.DEBIT,
                    value: 9000,
                }),
            );

            expect(result.current).toMatchSnapshot();
        });
        test('should call with entry (credit)', async () => {
            const { result } = renderHook(() =>
                useValue({
                    status: StatusEnum.COMPLETED,
                    entry: EntryEnum.CREDIT,
                    value: 9000,
                }),
            );

            expect(result.current).toMatchSnapshot();
        });
    });
});
