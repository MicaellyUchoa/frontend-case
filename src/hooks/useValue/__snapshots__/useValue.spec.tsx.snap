// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useValue Hook should call with status (refunded) 1`] = `
<p
  className="text-c_grayscale line-through sm:text-end"
>
  <span
    className="false"
  >
    R$ 90,00
  </span>
</p>
`;

exports[`useValue Hook useValue Hook (entry) should call with entry (credit) 1`] = `
<p
  className="text-c_secondary sm:text-end"
>
   + 
  <span
    className="font-black"
  >
    R$ 90,00
  </span>
</p>
`;

exports[`useValue Hook useValue Hook (entry) should call with entry (debit) 1`] = `
<p
  className="text-c_primary sm:text-end"
>
   - 
  <span
    className="font-black"
  >
    R$ 90,00
  </span>
</p>
`;
